addons/crm/wizard/crm_common_merge_opportunities.py

```diff
@@ -0,0 +1,76 @@
+        in a new opp).
+    """
+
+    _name = 'crm.common.merge.opportunity'
+    _description = 'Common Merge Opportunities'
+
+    active_id = fields.Many2one('crm.lead', readonly=True, default=lambda self: self.env.context.get('active_id'))
+    opportunity_ids = fields.Many2many('crm.lead', string='Leads/Opportunities')
+    merge_destination_opportunity_id = fields.Many2one('crm.lead', string="Destination opportunity",
+                                                       help="The opportunity from the list that will be used as destination. "
+                                                            "The information of the destination opportunity will be used in priority when merging")
+    user_id = fields.Many2one('res.users', 'Salesperson', index=True)
+    team_id = fields.Many2one('crm.team', 'Sales Team', oldname='section_id', index=True)
+
+    @api.onchange('opportunity_ids')
+    def _onchange_opportunity_ids(self):
+        """ When added/removing an opportunity, the merge_destination_opportunity_id should be the one
+            which has the highest confidence level
+        """
+        opportunities = self._get_highest_opportunity()
+        if opportunities:
+            self.merge_destination_opportunity_id = opportunities[0]
+            self.user_id = opportunities[0].user_id
+
+    @api.onchange('merge_destination_opportunity_id')
+    def _onchange_merge_destination_opportunity_id(self):
+        """ When modifying the destination opportunity, the user_id should be the one of the destination opportunity.
+            If no user is set on the destination opportunity, should be the one which has the highest confidence level
+        """
+        if self.merge_destination_opportunity_id and self.merge_destination_opportunity_id.user_id:
+            self.user_id = self.merge_destination_opportunity_id.user_id
+        else:
+            highest_opp = self._get_highest_opportunity()
+            if highest_opp:
+                self.user_id = highest_opp.user_id
+
+    @api.onchange('user_id')
+    def _onchange_user(self):
+        """ When changing the user, also set a team_id or restrict team id
+            to the ones user_id is member of.
+        """
+        if self.user_id:
+            if self.team_id:
+                user_in_team = self.env['crm.team'].search_count([('id', '=', self.team_id.id), '|', ('user_id', '=', self.user_id.id), ('member_ids', '=', self.user_id.id)])
+            else:
+                user_in_team = False
+            if not user_in_team:
+                values = self.env['crm.lead']._onchange_user_values(self.user_id.id if self.user_id else False)
+                self.team_id = values.get('team_id', False)
+
+    def _get_highest_opportunity(self):
+        """ Retrieves the opportunity that has the highest confidence level amongst the opportunity_ids. """
+        if self.opportunity_ids:
+            def opps_key(opportunity):
```

Comment:
maybe we should extract this from `merge_opportinity` and create a method on crm.lead. Will we get problem if one of the 2 implementations change ?